client/state/reader/feeds/actions.js

```diff
@@ -16,42 +15,34 @@ import {
 } from 'state/action-types';
 
 export function requestFeed( feedId ) {
-	return function( dispatch ) {
-		dispatch( {
-			type: READER_FEED_REQUEST,
-			payload: {
-				feed_ID: feedId,
-			},
-		} );
-		return wpcom
-			.undocumented()
-			.readFeed( { ID: feedId } )
-			.then(
-				function success( data ) {
-					dispatch( {
-						type: READER_FEED_REQUEST_SUCCESS,
-						payload: data,
-					} );
-					return data;
-				},
-				function failure( err ) {
-					dispatch( {
-						type: READER_FEED_REQUEST_FAILURE,
-						payload: {
-							feed_ID: feedId,
-						},
-						error: err,
-					} );
-					throw err;
-				}
-			);
+	return {
+		type: READER_FEED_REQUEST,
+		payload: {
+			ID: feedId,
+		},
+	};
+}
+
+export function receiveReaderFeedRequestSuccess( data ) {
+	return {
+		type: READER_FEED_REQUEST_SUCCESS,
+		payload: data,
+	};
+}
+
+export function receiveReaderFeedRequestFailure( action, error ) {
```

Comment:
could we pass less than the entire originating action? feels a little weird to pass so much when all we really want is the payload.

Reply:
the only piece that gets used in the reducer is `payload.feed_ID`, so yes